src/nvim/drawline.c

```diff
@@ -472,14 +472,14 @@ static void get_statuscol_display_info(LineDrawState *draw_state, int *char_attr
 {
   *c_extrap = NUL;
   *c_finalp = NUL;
+  const char *const text_end = stcp->text + stcp->text_len;
```

Comment:
LGTM, add this to the `statuscol_t` structure and set in `get_statuscol_str()` instead?

Reply:
I'll remove `text_len` then. And set `textp` to `NULL` to indicate that 'statuscolumn' hasn't been evaluated.